src/lib.rs

```diff
@@ -283,4 +285,31 @@ mod tests {
         }
         assert_eq!(iter.next(), None);
     }
+
+    fn random_unsigned_ints<T>()
+        where T: Unsigned + ToPrimitive + FromPrimitive + Ord + Clone
+            + std::fmt::Debug + rand::Rand {
+        const LENGTH: usize = 100000;
+        let mut v: Vec<T> = Vec::with_capacity(LENGTH);
+        for _ in 0..(LENGTH + 1) {
+            v.push(rand::random::<T>());
+        }
+        let v_len = v.len();
+        let packed_v = PackedVec::new(v.clone());
+        assert_eq!(packed_v.len(), v_len);
+        let mut iter = packed_v.iter();
+        for number in v {
```

Comment:
It might be more natural to use `zip` here. Something like `packed_v.iter().zip(v.iter()).all(|(x, y)| x == y)` (warning: totally untested).

Reply:
Ah, that *does* sound better!